pkgs/build-support/fetchgit/nix-prefetch-git

```diff
@@ -265,7 +265,9 @@ make_deterministic_repo(){
     rm -f .git/config
 
     # Garbage collect unreferenced objects.
-    git gc --prune=all
+    # Note: --keep-largest-pack prevents non-deterministic ordering of packs
+    #   listed in .git/objects/info/packs by only using a single pack
+    git gc --prune=all --keep-largest-pack
```

Comment:
An alternative suggested in #git was to just remove .git/objects/info/packs because this file seems to be only used for dumb http serves or so. That solution however would force an update to all existing hashes (with `keepDotGit = true`), so I didn't do that.

Reply:
Honestly, that sounds like the more correct solution for removing non-determinism. Maybe do a PR against staging?

Reply:
Well the problem isn't only hashes in nixpkgs, but also hashes for users outside nixpkgs, which is probably a whole lot more.

Reply:
This workaround will only affect derivations which were nondeterministic before

Reply:
You seem to be more confident in this area than I am. Maybe someone else will have a more informed opinion than me :)